File: framework/languages/java/e2e-kotlin/Monitoring/src/main/java/systems/zlink/e2e/kotlin/monitoring/Program.java
package systems.zlink.e2e.kotlin.monitoring;

public final class Program {
    private Program() {
    }

    public static void main(String[] args) {
        switch (Env.get("ZLINK_KOTLIN_E2E_ROLE", "client")) {
            case "registry" -> RegistryApplication.run(args);
            case "service" -> ServiceApplication.run(args);
            case "client" -> ClientApplication.run(args);
            case "validation" -> MonitoringValidationApplication.run(args);
            default -> throw new IllegalArgumentException(
                "unknown role " + Env.get("ZLINK_KOTLIN_E2E_ROLE"));
        }
    }
}

File: framework/languages/java/e2e-kotlin/Monitoring/src/main/kotlin/systems/zlink/e2e/kotlin/monitoring/Program.kt
package systems.zlink.e2e.kotlin.monitoring

fun main(args: Array<String>) {
    when (Env.get("ZLINK_KOTLIN_E2E_ROLE", "client")) {
        "registry" -> RegistryApplication.run(*args)
        "service" -> ServiceApplication.run(*args)
        "client" -> ClientApplication.run(*args)
        "validation" -> MonitoringValidationApplication.run(*args)
        else -> throw IllegalArgumentException("unknown role ${Env.get("ZLINK_KOTLIN_E2E_ROLE")}")
    }
}
